README.md

```diff
@@ -6,51 +6,51 @@ Reference implementation of Hex specifications: https://github.com/hexpm/specifi
 
 ## Usage
 
+Let's use default options for now. See "Configuration" section below for customization.
+
+```
+Options = hex_erl:default_options().
+```
+
 Get all package names:
 
 ```erlang
-hex_repo:get_names().
-%%=> {ok,
-%%=>     #{package => [
+hex_repo:get_names(Options).
+%%=> {ok, {200, ...,
+%%=>     #{packages => [
 %%=>         #{name => <<"package1">>},
 %%=>         #{name => <<"package2">>},
 %%=>         ...
-%%=>     ]},
-%%=>     [{etag, ...}, ...]}
+%%=>     ]}}}
 ```
 
-Note: By default, we fetch data from repo.hex.pm using built-in httpc-based adapter.
-See section below about configuring HTTP client.
-
 Get all package versions from repository:
 
 ```erlang
-hex_repo:get_versions().
-%%=> {ok,
+hex_repo:get_versions(Options).
+%%=> {ok, {200, ...,
 %%=>     #{packages => [
 %%=>         #{name => <<"package1">>, retired => [], versions => [<<"1.0.0">>]},
 %%=>         #{name => <<"package2">>, retired => [], versions => [<<"0.5.0">>]},
-%%=>     ]},
-%%=>     [{etag, ...}, ...]}
+%%=>     ]}}}
 ```
 
 Get package releases from repository:
 
 ```erlang
-hex_repo:get_package(<<"package1">>).
-%%=> {ok,
+hex_repo:get_package(<<"package1">>, Options).
+%%=> {ok, {200, ...,
 %%=>     #{releases => [
 %%=>         #{checksum => ..., version => <<"0.5.0">>, dependencies => []}],
 %%=>         #{checksum => ..., version => <<"1.0.0">>, dependencies => []}],
-%%=>     ]},
-%%=>     [{etag, ...}, ...]}
+%%=>     ]}}}
 ```
 
 Get package from HTTP API:
 
 ```erlang
-hex_api:get_package(<<"package1">>).
-%%=> {ok,
+hex_api:get_package(nil, <<"package1">>, Options).
```

Comment:
@ericmj see `hex_repo:get_package` vs `hex_api:get_package(Repo, <<"package1">>, Options)`; in the former I'm thinking we don't need to pass the organization name because it'll be part of the `repo_url` in the Options, in the latter `Repo` can be `nil` which means public repo or e.g. `acme` which means private. I'm thinking that the users of the library with multi repo support would always use the same base url for API but different urls for repos, like Hex does. I still think it's not ideal that functions have different signatures but I don't have a better idea at the moment, I think we can always revisit it later if it's worth pursuing. WDYT?

Reply:
I guess we could put optional `organization` key into options, some api functions will ignore it (eg around users) and some will take it (around packages)

Reply:
@ericmj I ended up putting `organization` key into options.

For now, I ended up with this section in README for configuring organizations:

---
Organizations on Hex.pm (or any compatible server) can be configured as following:

```erlang
Options = maps:merge(hex_erl:default_options(), #{
    api_key => APIKey,
    organization => <<"acme">>,
    repo_key => RepoKey,
    repo_uri => <<"https://repo.hex.pm/repos/acme">>
}).
```
---

WDYT? I think we always have different key per authed private repo, right?